docs/pages/connect-your-client/gui-clients.mdx

```diff
@@ -501,31 +501,160 @@ Click "Proceed", then click "Finish".
+| Connection type | Basic                  |
+| Host name       | `localhost`            |
+| Port number     | `--port` flag value    |
+| Type            | Service Name           |
+| Service name    | `--db-name` flag value |
+
+Test and create the connection.
+
+![SQL Developer (VS Code) Connection Details (basic)](../../img/database-access/guides/oracle/sql-developer-vscode-conn-details-basic@2x.png)
+
+The new connection should appear on the list.
+
+<Figure width="400">
+![SQL Developer (VS Code) Connected (basic)](../../img/database-access/guides/oracle/sql-developer-vscode-connected-basic@2x.png)
+</Figure>
+
+</TabItem>
+
+<TabItem label="TCPS">
+
+Open the extension toolbar and click on "Create Connection" button.
+
+![SQL Developer (VS Code) Start](../../img/database-access/guides/oracle/sql-developer-vscode-start@2x.png)
+
+Enter the following connection details:
+
+| Field            | Value                           |
+|------------------|---------------------------------|
+| Connection name  | (choose per your preference)    |
+| User name        | `/`                             |
+| Password         | `/`                             |
+| Save Password    | Mark the checkbox               |
+| Connection type  | "Custom JDBC"                   |
+| Custom JDBC URL  | Copy from `tsh proxy db` output |
+
+Test and create the connection.
+
+![SQL Developer (VS Code) Connection Details (JDBC)](../../img/database-access/guides/oracle/sql-developer-vscode-conn-details-jdbc@2x.png)
+
+The new connection should appear on the list.
+
+<Figure width="400">
+![SQL Developer (VS Code) Connected (JDBC)](../../img/database-access/guides/oracle/sql-developer-vscode-connected-jdbc@2x.png)
+</Figure>
+
+
+</TabItem>
+</Tabs>
+
+### Toad
+
+Add new login record in the logins dialog.
+
+![Toad Add Login Record](../../img/database-access/guides/oracle/toad-add-login-record@2x.png)
+
+Enter the connection details in "Direct" tab:
+
+| Field           | Value                        |
+|-----------------|------------------------------|
+| Host name       | `127.0.0.1`                  |
+| Port number     | `--port` flag value          |
+| Service name    | `--db-name` flag value       |
+| User name       | `EXTERNAL`                   |
+| Password        | (leave empty)                |
+| Connection name | (choose per your preference) |
+
+Test the connection to verify the setup.
+
+![Toad Add Login Tested](../../img/database-access/guides/oracle/toad-add-login-tested@2x.png)
+
+The newly added login should appear on the login list.
+
+![Toad Login List](../../img/database-access/guides/oracle/toad-login-list@2x.png)
+
+<Notice type="tip">
+You can also configure Toad to use an external Oracle client. Both native and external clients are supported.
+</Notice>
+
+### DBeaver
+
+Click on the "New Database Connection" button.
+
+![DBeaver new connection button](../../img/database-access/guides/oracle/dbeaver-new-connection@2x.png)
```

Comment:
I allowed myself to run images from this PR through [ImageOptim](https://imageoptim.com/mac). This reduced their size by half, without reducing quality.

<img width="549" alt="imageoptim" src="https://github.com/user-attachments/assets/6692c4bc-d8dd-4212-8657-1877f641358f" />

I also added `@2x` suffix to the new images that are high DPI. This makes the doc engine handle them better when you click on an image. Make your browser fullscreen and then compare [the old version](https://docs-lxv4wvpbr-goteleport.vercel.app/docs/connect-your-client/gui-clients/#dbeaver) vs the new version by clicking on an image.

Reply:
Thank you!